packages/react-storage/src/components/StorageBrowser/Views/LocationsView/Controls/DataTable.tsx

```diff
@@ -0,0 +1,156 @@
+import React from 'react';
+
+import { capitalize } from '@aws-amplify/ui';
+
+import {
+  DataTable,
+  TABLE_DATA_BUTTON_CLASS,
+  TABLE_HEADER_BUTTON_CLASS_NAME,
+  TABLE_HEADER_CLASS_NAME,
+} from '../../../components/DataTable';
+import { useControl } from '../../../context/controls';
+import { useLocationsData } from '../../../context/actions';
+import { LocationAccess } from '../../../context/types';
+import { compareStrings } from '../../../context/controls/Table';
+import { ButtonElement, IconElement } from '../../../context/elements';
+
+export type SortDirection = 'ascending' | 'descending' | 'none';
+
+export type SortState = {
+  selection: string;
+  direction: SortDirection;
+};
+
+const getCompareFn = (selection: string) => {
+  switch (selection) {
+    case 'scope':
+    case 'type':
+    case 'permission':
+      return compareStrings;
+  }
+};
+
+const getColumnItem = ({
+  entry,
+  selection,
+  direction,
+  onTableHeaderClick,
+}: {
+  entry: [string, string];
+  selection: string;
+  direction: SortDirection;
+  onTableHeaderClick: (location: string) => void;
+}) => {
+  const [key, value] = entry;
+
+  return {
+    children: (
+      <ButtonElement variant="sort" className={TABLE_HEADER_BUTTON_CLASS_NAME}>
+        {capitalize(value)}
+        <IconElement
+          variant={
+            selection === key && direction !== 'none'
+              ? `sort-${direction}`
+              : 'sort-indeterminate'
+          }
+        />
+      </ButtonElement>
+    ),
+    key: `th=${key}`,
```

Comment:
Why `=` here instead of `_`? Can we align on a syntax?

Reply:
ah that was a typo, will update